### PR TITLE
add mkdocstrings

### PR DESCRIPTION
**Pull request by bandersen23**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by bandersen23** [comment]
@conda-forge/help-python This should be good to go! Thank you!

File: recipes/mkdocstrings/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  host:
+    - pip
+    - poetry
+    - python >=3.6,<4.0
+  run:
+    - beautifulsoup4 >=4.8.2,<5.0.0
+    - mkdocs >=1.1,<2.0
+    - pymdown-extensions >=6.3,<9.0
+    - python >=3.6,<4.0
+    - pytkdocs >=0.2.0,<0.10.0
```
Comment (beckermr) [inline recipes/mkdocstrings/meta.yaml:28]: I am happy with these constraints, but if you can loosen any of them, I'd recommend it.

Comment (beckermr) [reply recipes/mkdocstrings/meta.yaml:28]: Except for python. Leave that one as is.

Comment (bandersen23) [reply recipes/mkdocstrings/meta.yaml]: @beckermr I adjusted those to take better match the specification. Good catch!

File: recipes/mkdocstrings/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    - mkdocstrings.handlers
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://github.com/pawamoy/mkdocstrings
+  summary: Automatic documentation from sources, for MkDocs.
+  license: MIT
```
Comment (pawamoy) [inline recipes/mkdocstrings/meta.yaml:42]: This should be ISC :slightly_smiling_face:

**Comment by bandersen23** [comment]
@pawamoy I fixed this in conda-forge/mkdocstrings-feedstock#3 - it should be good to go there! I did the same with pytkdocs.

**Comment by pawamoy** [comment]
Alright, thanks @bandersen23! I wasn't how these recipes and feedstocks work :smile:

**Comment by bandersen23** [comment]
@pawamoy This was just the initial addition to conda-forge then all other updates happen in the `conda-forge/mkdocstrings-feedstock` repo. Would you like to be added as a maintainer of that repo?

**Comment by pawamoy** [comment]
If you feel this adds value, why not, but I barely know conda :sweat_smile:

**Pull request merged**